dashboard/graphql/objects.py

```diff
@@ -0,0 +1,221 @@
+            'assignment_group_id': parent.id,
+            'id': assignment_id,
+        })
+
+    class Meta:
+        model = AssignmentGroups
+        only_fields = (
+            'id', 'course_id', 'name', 'weight', 'group_points', 'drop_lowest', 'drop_highest'
+        )
+
+
+class AssignmentType(DjangoObjectType):
+    id = graphene.ID()
+    name = graphene.String()
+    due_date = graphene.types.datetime.DateTime()
+    local_date = graphene.types.datetime.DateTime()
+    points_possible = graphene.Float()
+    course_id = graphene.ID()
+    assignment_group_id = graphene.ID()
+    average_grade = graphene.Float()
+    median_grade = graphene.Float()
+
+    submissions = graphene.List(SubmissionType)
+    current_user_submission = graphene.Field(SubmissionType)
+
+    assignment_group = graphene.Field(AssignmentGroupType)
+
+    def resolve_submissions(parent, info):
+        user = info.context.user
+        if not is_admin_or_instructor_in_course_id.test(user, parent.course_id):
+            raise GraphQLError('You do not have permission to access this resource!')
+
+        return info.context.submissions_by_assignment_id_loader.load(parent.id)
+
+    def resolve_current_user_submission(parent, info):
+        canvas_user_id = info.context.canvas_user_id
+
+        return info.context.submission_by_assignment_id_and_user_id_loader.load({
+            'assignment_id': parent.id,
+            'user_id': canvas_user_id,
+        })
+
+    def resolve_assignment_group(parent, info):
+        return info.context.assignment_group_by_course_id_and_id_loader.load({
+            'course_id': parent.course_id,
+            'id': parent.assignment_group_id,
+        })
+
+    def _average_grade_lambda(parent, info, submissions):
+        if len(submissions) > 0:
+            return np.average([submission.score if submission.score else 0 for submission in submissions])
+        return 0
+
+    def resolve_average_grade(parent, info):
+        return info.context.submissions_by_assignment_id_loader.load(parent.id).then(
+            lambda submissions: AssignmentType._average_grade_lambda(parent, info, submissions)
+        )
+
+    def _median_grade_lambda(parent, info, submissions):
+        if len(submissions) > 0:
+            return np.median([submission.score if submission.score else 0 for submission in submissions])
+        return 0
+
+    def resolve_median_grade(parent, info):
+        return info.context.submissions_by_assignment_id_loader.load(parent.id).then(
+            lambda submissions: AssignmentType._median_grade_lambda(parent, info, submissions)
+        )
+
+    class Meta:
+        model = Assignment
+        only_fields = (
+            'id', 'name', 'due_date', 'local_date', 'points_possible', 'course_id', 'assignment_group_id'
+        )
+AssignmentGroupType.assignments = graphene.List(AssignmentType)
+AssignmentGroupType.assignments = graphene.List(AssignmentType)
```

Comment:
Can someone explain why this is repeated and placed here specifically?

Reply:
@andrew-gardener?